src/main.rs

```diff
@@ -79,10 +85,47 @@ fn create_and_add_row(path: String, database: &Connection) -> anyhow::Result<Str
     add_stmt.execute([&uuid, &path, &time])?;
     Ok(uuid)
 }
-async fn redirect(State(shared_state): State<AppState>, data_string: String) -> Response {
+
+async fn redirect_post_mulipart_form(
+    State(shared_state): State<AppState>,
+    mut multipart: Multipart,
+) -> Response {
     let database: tokio::sync::MutexGuard<'_, Connection> = shared_state.database.lock().await;
+    let mut data_array: Vec<File> = Vec::new();
+    let mut ifcfg = false;
+    let mut xml = false;
+    while let Some(field) = multipart.next_field().await.unwrap() {
+        let file_type = field.content_type().unwrap().to_string();
+        let file_name = field.file_name().unwrap().to_string();
+
+        let data = field.bytes().await.unwrap();
+
+        let file_content = match str::from_utf8(&data) {
+            Ok(v) => v.to_string(),
+            Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+        };
+
+        if file_type == "text/plain" {
+            ifcfg = true
+        };
+        if file_type == "text/xml" {
+            xml = true
+        };
```

Comment:
I would get rid of the `ifcfg` and `xml` variables. My approach would be to create an enum with the supported file types which implements a `TryFrom` for the content_type strings and then store it in the `File` struct. For now storing it isn't that important but in the future it probably makes sense to support netconfig so a third type which would then need to be identifyable by `migrate()` (because it needs to be at `/etc/sysconfig/network/config` in the container).